parsers/MX.py

```diff
@@ -22,6 +31,115 @@
              "BZ->MX-PN": "IntercambioPEN-BEL"
              }
 
+MAPPING = {
+    "Eolica": "wind",
+    "Fotovoltaica": "solar",
+    "Biomasa": "biomass",
+    "Carboelectrica": "coal",
+    "Ciclo Combinado": "gas",
+    "Geotermoelectrica": "geothermal",
+    "Hidroelectrica": "hydro",
+    "Nucleoelectrica": "nuclear",
+    "Combustion Interna": "unknown",
+    "Termica Convencional": "unknown",
+    "Turbo Gas": "gas"
+}
+
+# cache where the data for whole months is stored as soon as it has been fetched once
+DATA_CACHE = {}
+
+
+def parse_date(date, hour):
+    tzoffset = tz.tzoffset("CDT", -3600 * 6)
```

Comment:
I do not know whether that is correct also in case of daylight saving time

Reply:
I could not find a source stating which time zone the data is reported in but I assume that is Central Standard Time (CST) where Mexico City belongs to, which is UTC-6. There are no holes in the data when daylight saving time changes happened so I guess that daylight saving time is never reported.